docs/tokens/tutorials/validate-1sat-input-with-oracle.md

```diff
@@ -0,0 +1,129 @@
+---
+sidebar_position: 5
+---
+
+# Tutorial 5: Ordinals Oracle
+
+## Overview
+Bitcoin smart contracts can decide whether a UTXO is valid, but cannot directly determine whether the [1SatOrdinals](https://docs.1satordinals.com/) tokens in a UTXO are valid, since they are validated by an external indexer off chain outside of miners. In many practical applications, verifying the Ordinals tokens carried in certain transaction inputs is necessary, such as token swap and token sale. [Oracles](../../tutorials/oracle.md) must be introduced to provide additional verification for the authenticity and integrity of the Ordinals tokens required when calling a contract.
+
+This tutorial will introduce how to use the [WitnessOnChain](https://api.witnessonchain.com) oracle to validate transaction inputs referencing UTXOs containing Ordinals NFTs and BSV20 tokens.
+
+## WitnessOnChain API
+
+WitnessOnChain provides an [API](https://api.witnessonchain.com/#/v1/V1Controller_getInscription) to get inscription details from an outpoint.
+
+```
+https://api.witnessonchain.com/v1/inscription/bsv/{network}/outpoint/{txid}/{vout}
+```
+
+The structure of the signed message in response is as follows:
+
+| Name      | Type       | Bytes | Description                                  |
+| --------- | ---------- | ----- | -------------------------------------------- |
+| marker    | bigint     | 1     | api marker, always be 4n                     |
+| timestamp | bigint     | 4     | timestamp, little-endian                     |
+| network   | bigint     | 1     | network type, 1n for mainnet, 0n for testnet |
+| txid      | ByteString | 32    | txid                                         |
+| vout      | bigint     | 4     | output index, little-endian                  |
```

Comment:
Merge these two into `outpoint`, so we do not have to compare separate to what's in `this.prevouts`. They are always used together.